From d3c7585c25de4f91b754aed86a69d0629a643c5e Mon Sep 17 00:00:00 2001
Message-ID: <d3c7585c25de4f91b754aed86a69d0629a643c5e.1734811274.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1734811274.git.sharlatanus@gmail.com>
References: <cover.1734811274.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Fri, 20 Dec 2024 23:33:02 +0000
Subject: [PATCH 35/38] gnu: python-specutils: Fix build.

* gnu/packages/astronomy.scm (python-specutils):
[arguments] <test-flags>: Skip one more test.
<phases>: Remove 'relax-requirements.

Change-Id: Iba3837d8d87ece1e97f954ffb20afdbfd6e05654
---
 gnu/packages/astronomy.scm | 24 +++++++++++++-----------
 1 file changed, 13 insertions(+), 11 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index e368ce57d4..d180ea4451 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -4950,25 +4950,27 @@ (define-public python-specutils
     (build-system pyproject-build-system)
     (arguments
      (list
-      ;; Disabling test requiring access to download
-      ;; <https://datacenter.iers.org/data/9/finals2000A.all>.
-      ;; XXX: Check if test data may be packed as standalone package.
-      #:test-flags #~(list "-k" "not test_create_spectral_axis")
+      #:test-flags
+      #~(list "-k" (string-join
+                    (list
+                     ;; Disabling test requiring access to download
+                     ;; <https://datacenter.iers.org/data/9/finals2000A.all>.
+                     ;; XXX: Check if test data may be packed as standalone
+                     ;; package.
+                     "not test_create_spectral_axis"
+                     ;; ValueError: Expected the following order of world
+                     ;; arguments: SpectralCoord
+                     "test_wcs_transformations")
+                    " and not "))
       #:phases
       #~(modify-phases %standard-phases
-          (add-after 'unpack 'relax-requirements
-            (lambda _
-              (substitute* "setup.cfg"
-                ;; numpy>=1.24
-                (("1.24") "1.23.2"))))
           (add-before 'check 'set-home-env
             (lambda _
               ;; Tests require HOME to be set.
               ;;  Permission denied: '/homeless-shelter'
               (setenv "HOME" "/tmp"))))))
     (propagated-inputs
-     (list ;; python-stdatamodels ; cycle with python-synphot, optional.
-           python-asdf
+     (list python-asdf
            python-asdf-astropy
            python-astropy
            python-gwcs
-- 
2.46.0

